### PR TITLE
Upgrade project to support EPiServer 12

### PR DESCRIPTION
**Pull request by pciarach**

### CODE REVIEW

**Comment by pciarach** [comment]
Temporarily restored AlloyDemoKit so that the diff panel is not freezing... I'm going to remove it just before merging

File: EpiserverRedirects/System/SystemRedirectsActions.cs [hunk 1]
```diff
@@ -4,23 +4,31 @@
...
 {
-    public static class SystemRedirectsActions
+    public class SystemRedirectsActions
     {
-        public static void AddRedirects(PageData pageData, string oldUrl, CultureInfo cultureInfo,
-            SystemRedirectReason systemRedirectReason)
+        private readonly IContentRepository _contentRepository;
+        private readonly IRedirectRuleRepository _redirectRuleRepository;
+
+        public SystemRedirectsActions(IContentRepository contentRepository, IRedirectRuleRepository redirectRuleRepository)
```
Comment (tomaszmadeyski) [inline EpiserverRedirects/System/SystemRedirectsActions.cs:18]: It seems like IContentLoader is enough (we don't need IContentRepository)

File: EpiserverRedirects/EpiserverRedirects.csproj [hunk 2]
```diff
@@ -1,86 +1,50 @@
...
+    <PropertyGroup>
+        <ProjectDir Condition="'$(ProjectDir)' == ''">$(MSBuildProjectDirectory)</ProjectDir>
+        <SolutionDir Condition="'$(SolutionDir)' == ''">$(ProjectDir)</SolutionDir>
+        <TmpOutDir Condition="'$(TmpOutDir)' == ''">$(OutputPath)/temp</TmpOutDir>
+        <!-- NOTE! This is different between cms versions, the zip file needs to be named the same in the package -->
+        <ZipFile>$(OutputPath)\Forte.EpiserverRedirects.zip</ZipFile>
+        <RunPostBuildEvent>Always</RunPostBuildEvent>
+    </PropertyGroup>
+    <Target Name="CreateNugetPackage" AfterTargets="Build;Rebuild" Condition="'$(Configuration)' == 'Release' ">
+        <Message Text="Creating Nuget Package" />
```
[the remark this answers is not in the document]
Comment (pciarach) [reply EpiserverRedirects/EpiserverRedirects.csproj:35]: Yes they are required in order to prepare a zip file containing module-related files that will be unpacked after installing NuGet

File: EpiserverRedirects.Tests/Data/RedirectRuleTestDataBuilder.cs [hunk 3]
```diff
@@ -69,7 +69,9 @@ private RedirectRule ChangeData(Action<RedirectRule> changeDataAction)
                 var randomRedirectRule = GetRandomRedirectRuleFromData();
 
                 if (_alreadyChangedRedirectRulesGuids.Contains(randomRedirectRule.Id.ExternalId))
```
Comment (smydolf) [inline EpiserverRedirects.Tests/Data/RedirectRuleTestDataBuilder.cs:71]: why this line is formatted in this way?

File: EpiserverRedirects/Caching/ICache.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Forte.EpiserverRedirects.Caching
+{
+    public interface ICache : ICacheRemover
+    {
+        bool TryGet<T>(string key, out T resource) where T : class;
```
Comment (smydolf) [inline EpiserverRedirects/Caching/ICache.cs:5]: is resource equals redirect (from method Add)? If yes could you adjust the naming here?

File: EpiserverRedirects/ClientResources/RedirectsMenu/RedirectsMenuGrid/RedirectsMenuGrid.css [hunk 5]
```diff
@@ -17,11 +17,12 @@
...
 
 .url-redirect-grid-container .dgrid .dgrid-scroller {
-    margin-top: 55px;
+    margin-top: 100px !important;
+    margin-bottom: 30px !important;
 }
 
 .url-redirect-grid-container .dgrid .dgrid-header {
-    height: 55px;
+    height: 100px !important;
```
Comment (smydolf) [inline EpiserverRedirects/ClientResources/RedirectsMenu/RedirectsMenuGrid/RedirectsMenuGrid.css:25]: image

**Review by smydolf — APPROVED** [review]

File: EpiserverRedirects/ClientResources/RedirectsMenu/RedirectsMenuGrid/RedirectsMenuGrid.css (hunk 5 above)
Comment (pciarach) [reply EpiserverRedirects/ClientResources/RedirectsMenu/RedirectsMenuGrid/RedirectsMenuGrid.css:25]: Unfortunately, the header receives inline styles when sorting by column so it was kind of the cheapest solution...
I really, really regret causing this death

**Review by tomaszmadeyski — APPROVED** [review]

**Pull request merged**